File: inbloom-hello-world-ruby/app/controllers/recipes/recipe2.rb
# Determine user’s educational organization id
def recipe2()
  return getEduOrgs()
end

def getEduOrgs()
  # get the home endpoint
	data = JSON.parse RestClient.get REST_URL + "home", headers
	links = data['links']
	

  # find the link to the Edu Orgs	
	eduOrgs_url = links.each do |link| 
	      if link['rel'] == 'getEducationOrganizations'
 					break link['href'].to_s
 			  end
  end

  # get the edu orgs	
	eduOrgs = JSON.parse RestClient.get eduOrgs_url, headers

  # pull out the ids and return then	
	return eduOrgs_url, eduOrgs.map {|eduOrg| eduOrg['id'] }
  
end

File: hello-world-c/inBloom-c-sharp-hello-world/Recipes/Recipe2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Web;
using System.Web.Configuration;
using RestSharp;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;
using InBloomClient.Entities;

namespace inBloom_c_sharp_hello_world.Recipes
{

    // Determine user's educational organization ids
    public class Recipe2 : Recipe
    {
        /// <summary>
        /// Run the current recipe
        /// </summary>
        /// <param name="accessToken">
        /// The access token.
        /// </param>
        /// <returns>
        /// A concatenated string of all of the IDs of the Education Organizations the user is associated with
        /// </returns>
        override public string RunRecipe(string token)
        {
            string ret = "";
            
            IEnumerable<EducationOrganization> eduOrgs = getEducationalOrgs(token);

            // pull out the id fields and return it
            foreach (EducationOrganization eduOrg in eduOrgs)
            {
                ret += eduOrg.id + "\t";
            }
            return ret;
        }

        public IEnumerable<EducationOrganization> getEducationalOrgs(string token)
        {
            var client = new RestClient(WebConfigurationManager.AppSettings["inBloomRestURL"]);
            string homeEndpoint = "home/";

            // retrieve the home endpoint
            RestRequest homeRequest = inBloomRestRequest(token, homeEndpoint, Method.GET);
            var homeResponse = client.Execute(homeRequest);
            dynamic home = JsonConvert.DeserializeObject(homeResponse.Content);

            // find the self endpoint's url
            JArray links = home["links"];
            string eduOrgsEndpoint = findLink("getEducationOrganizations", links).Replace(WebConfigurationManager.AppSettings["inBloomRestURL"], "");

            // retrieve the self endpoint
            RestRequest eduOrgsRequest = inBloomRestRequest(token, eduOrgsEndpoint, Method.GET);
            var eduOrgsResponse = client.Execute(eduOrgsRequest);
            return JsonConvert.DeserializeObject<List<EducationOrganization>>(eduOrgsResponse.Content);
        }       
    }
}